services/diddoc/diddoc_query.go

```diff
@@ -23,34 +29,283 @@ func (dd *QueryDIDDocRequestService) SpecificValidation(c services.ResolverConte
+		if err != nil {
+			return err
+		}
 	}
+
+	dd.FirstHandler = &startHandler
+
 	return nil
 }
+
+func (dd *QueryDIDDocRequestService) RegisterDidDocQueryHanlders(startHandler queries.BaseQueryHandlerI, c services.ResolverContext) (queries.BaseQueryHandlerI, error) {
+	// - didQueryHandler
+	// or
+	// - versionIdHandler
+	// After that we can find for service field if it's set.
+	// VersionIdHandler -> VersionTimeHandler -> DidDocResolveHandler -> DidDocMetadataHandler -> ServiceHandler -> RelativeRefHandler
+	relativeRefHandler := diddocQueries.RelativeRefHandler{}
+	serviceHandler := diddocQueries.ServiceHandler{}
+	versionIdHandler := diddocQueries.VersionIdHandler{}
+	versionTimeHandler := diddocQueries.VersionTimeHandler{}
+	didDocResolveHandler := diddocQueries.DidDocResolveHandler{}
+	didDocMetadataHandler := diddocQueries.DidDocMetadataHandler{}
+
+	err := startHandler.SetNext(c, &versionIdHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	err = versionIdHandler.SetNext(c, &versionTimeHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	err = versionTimeHandler.SetNext(c, &didDocResolveHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	err = didDocResolveHandler.SetNext(c, &didDocMetadataHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	err = didDocMetadataHandler.SetNext(c, &serviceHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	err = serviceHandler.SetNext(c, &relativeRefHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	return &relativeRefHandler, nil
+}
+
+func (dd *QueryDIDDocRequestService) RegisterResourceQueryHandlers(startHandler queries.BaseQueryHandlerI, c services.ResolverContext) (queries.BaseQueryHandlerI, error) {
+	// Resource handlers
+	resourceQueryHandler := resourceQueries.ResourceQueryHandler{}
+	resourceIdHandler := resourceQueries.ResourceIdHandler{}
+	resourceMetadataHandler := resourceQueries.ResourceMetadataHandler{}
+	resourceCollectionIdHandler := resourceQueries.ResourceCollectionIdHandler{}
+	resourceNameHandler := resourceQueries.ResourceNameHandler{}
+	resourceTypeHandler := resourceQueries.ResourceTypeHandler{}
+	resourceVersionHandler := resourceQueries.ResourceVersionHandler{}
+	resourceVersionTimeHandler := resourceQueries.ResourceVersionTimeHandler{}
+	resourceValidationHandler := resourceQueries.ResourceValidationHandler{}
+	resourceChecksumHandler := resourceQueries.ResourceChecksumHandler{}
+
+	err := startHandler.SetNext(c, &resourceQueryHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	// It's a resource query to fetch the collection of resources
+	err = resourceQueryHandler.SetNext(c, &resourceIdHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	// Resource handlers
+	// Chain would be:
+	// resourceIdHandler -> resourceCollectionIdHandler ->
+	// -> resourceNameHandler -> resourceTypeHandler -> resourceVersionHandler ->
+	// -> resourceVersionTimeHandler -> resourceValidationHandler -> resourceMetadataHandler -> stopHandler
```

Comment:
Let's add in that comment another handlers too:
- `resourceQueryHandler`
- `resourceChecksumHandler`
- `resourceMetadataHandler`